cmd/juju/commands/bootstrap.go

```diff
@@ -562,6 +574,38 @@ to create a new model to deploy k8s workloads.
 		return errors.Trace(err)
 	}
 
+	// If region is specified by the user, validate it here.
+	// lp#1632735
+	if c.Region != "" {
+		_, err := jujucloud.RegionByName(cloud.Regions, c.Region)
+		if err != nil {
+			allRegions := make([]string, len(cloud.Regions))
+			for i, one := range cloud.Regions {
+				allRegions[i] = one.Name
+			}
+			if len(allRegions) > 0 {
+				naturalsort.Sort(allRegions)
+				plural := "s are"
+				if len(allRegions) == 1 {
+					plural = " is"
+				}
+				ctx.Infof("Available cloud region%v %v", plural, strings.Join(allRegions, ", "))
+			}
+			return errors.NotValidf("region %q for cloud %q", c.Region, c.Cloud)
+		}
+	}
+
+	isCAASController = jujucloud.CloudIsCAAS(cloud)
```

Comment:
this could be moved closer to line 649 where it's used

Reply:
Done.